README.md

```diff
@@ -276,21 +276,37 @@ package flatend
 import "github.com/lithdew/kademlia"
 
 type Node struct {
-    // A reachable, public address which peers may reach you on.
-    // The format of the address must be [host]:[port].
-    PublicAddr string
+	// A reachable, public address which peers may reach you on.
+	// The format of the address must be [host]:[port].
+	PublicAddr string
 
-    // A 32-byte Ed25519 private key. A secret key must be provided
-    // to allow for peers to reach you. A secret key may be generated
-    // by calling `flatend.GenerateSecretKey()`.
-    SecretKey kademlia.PrivateKey
+	// A 32-byte Ed25519 private key. A secret key must be provided
+	// to allow for peers to reach you. A secret key may be generated
+	// by calling `flatend.GenerateSecretKey()`.
+	SecretKey kademlia.PrivateKey
 
 	// A list of IPv4/IPv6 addresses and ports assembled as [host]:[port] which
 	// your Flatend node will listen for other nodes from.
-    BindAddrs []string
+	BindAddrs []string
 
-    // A mapping of service names to their respective handlers.
-    Services map[string]Handler
+	// A mapping of service names to their respective handlers.
+	Services map[string]flatend.Handler
+
+	// Total number of attempts to reconnect to a peer we reached that disconnected.
+	// Default is 8 attempts, set to a negative integer to not attempt to reconnect at all.
```

Comment:
Can 0 be no reconnect and -1 be forever retry?

Reply:
I tried to figure a way to do that, though I wonder how we could force the default value for NumReconnectAttempts to be 8 in that case. The only idiomatic way I could think of is to switch to functional options, or to encapsulate options into a separate Options struct with sensible defaults, though might you have any suggestions?

Reply:
When I do services in Go, typically I create some kind of config object which can be use by `main.go`. So the main program is the one that is setting the defaults. User can then set parameters on `main.go` to override the defaults.

Reply:
@lithdew Why did you choose 8 as default?